### PR TITLE
Add LSP-robloxlua package

### PR DESCRIPTION
**Pull request by alihsaas**

### CODE REVIEW

File: repository/l.json [hunk 1]
```diff
@@ -1717,6 +1717,19 @@
...
 		},
+		{
+			"name": "LSP-luau",
+			"details": "https://github.com/alihsaas/LSP-luau",
+			"labels": [
+				"lsp", "roblox", "luau"
+			],
+			"releases": [
+				{
+					"sublime_text": ">=3092",
```
Comment (rchl) [inline repository/l.json:1728]: Your package is only compatible with ST4 since you are using LSP API that is only available there.

**Comment by rchl** [comment]
This is based on https://github.com/sublimelsp/LSP-lua right? How do those two compare?

File: repository/l.json (hunk 1 above)
Comment (alihsaas) [reply repository/l.json:1728]: This uses the server provided in https://github.com/NightrainsRbx/RobloxLsp which adds support for Roblox's Luau

**Comment by alihsaas** [comment]
> 
> 
> This is based on https://github.com/sublimelsp/LSP-lua right? How do those two compare?

This uses the server provided in https://github.com/NightrainsRbx/RobloxLsp which adds support for Roblox's Luau

**Comment by rchl** [comment]
> > This is based on [sublimelsp/LSP-lua](https://github.com/sublimelsp/LSP-lua) right? How do those two compare?
> 
> This uses the server provided in [NightrainsRbx/RobloxLsp](https://github.com/NightrainsRbx/RobloxLsp) which adds support for Roblox's Luau

I wonder if we should consolidate those somehow because the user will likely not understand the difference between two almost alike-named servers.

**Comment by alihsaas** [comment]
> 
> I wonder if we should consolidate those somehow because the user will likely not understand the difference between two almost alike-named servers.
Would LSP-RobloxLuau work?

**Comment by rwols** [comment]
I think it's okay to have two different lua language servers. I don't program in lua myself, but from what I understand searching on the internet there seem to be many dialects of this language:

- The base "lua" language
- A dialect from "Roblox"
- GLua from Garry's Mod
- EmmyLua for type annotations (?) https://github.com/EmmyLua/EmmyLua-LanguageServer

There's a bit of a problem with respect to the syntaxes/languageIDs: if a user has both the base lua language server as well as the roblox lua language server, then they'll both be attached to the same file. How do people handle this in vscode land?

There's opportunity to improve your repository, @alihsaas. See my patch notes: https://github.com/sublimelsp/LSP-lua/blob/master/patch-notes

**Comment by alihsaas** [comment]
You can read more on Roblox's Luau here https://roblox.github.io/luau/

> 
> There's a bit of a problem with respect to the syntaxes/languageIDs: if a user has both the base lua language server as well as the roblox lua language server, then they'll both be attached to the same file. How do people handle this in vscode land?

I don't know, but I have added a syntax definition file, doesn't include type syntax if that helps.

> 
> There's opportunity to improve your repository, @alihsaas. See my patch notes: https://github.com/sublimelsp/LSP-lua/blob/master/patch-notes

Applied the patch. 👍

**Comment by rwols** [comment]
The amount of LSP-* packages is getting out of hand, so we've made our own package repository. Please submit your pull request there: https://github.com/sublimelsp/repository/

**Pull request closed without merging**